libs/template/class/src/lib/class.directive.ts

```diff
@@ -0,0 +1,197 @@
+import {
+  Directive,
+  ElementRef,
+  Input,
+  IterableChanges,
+  IterableDiffer,
+  IterableDiffers,
+  KeyValueChanges,
+  KeyValueDiffer,
+  KeyValueDiffers,
+  OnDestroy,
+  OnInit,
+  Output,
+  Renderer2,
+  ɵisListLikeIterable as isListLikeIterable
+} from '@angular/core';
+import {
+  accumulateObservables,
+  RxStrategyNames,
+  RxStrategyProvider
+} from '@rx-angular/cdk';
+import { coerceAllFactory } from '@rx-angular/cdk/coercing';
+import {
+  BehaviorSubject,
+  isObservable,
+  Observable,
+  ObservableInput,
+  ReplaySubject,
+  Subject,
+  Subscription
+} from 'rxjs';
+import { map, switchAll, switchMap, withLatestFrom } from 'rxjs/operators';
+
+type NgClassValues =
+  | string[]
+  | Set<string>
+  | Record<string, boolean>
+  | string
+  | null
+  | undefined;
+
+type RxClassInput =
+  | Observable<NgClassValues>
+  | Record<string, Observable<boolean>>;
+
+@Directive({ selector: '[rxClass]' })
+export class ClassDirective implements OnInit, OnDestroy {
+  private iterableDiffer: IterableDiffer<string> | null = null;
+  private keyValueDiffer: KeyValueDiffer<string, any> | null = null;
+  private rawClass: NgClassValues = null;
+  private newInput = false;
+  private sub = new Subscription();
+  private classValueChanged = false;
+
+  constructor(
+    private readonly iterableDiffers: IterableDiffers,
+    private readonly keyValueDiffers: KeyValueDiffers,
+    private readonly ngEl: ElementRef,
+    private readonly renderer: Renderer2,
+    private readonly strategyProvider: RxStrategyProvider
+  ) {}
+
+  private rxClass = coerceAllFactory<NgClassValues>(
+    () => new ReplaySubject(1),
+    switchAll()
+  );
+
+  @Input('rxClass')
+  set rxClass$(value$: RxClassInput) {
+    this.newInput = true;
+    this.rxClass.next(
+      (isObservable(value$) ? value$ : accumulateObservables(value$)).pipe(
+        map((value) => (typeof value === 'string' ? [value] : value))
+      )
+    );
+  }
+
+  private strategy = coerceAllFactory(
+    () => new BehaviorSubject<ObservableInput<string> | string>(this.strategyProvider.primaryStrategy),
+    switchAll()
+  );
+
+  @Input('rxClassStrategy')
+  set strategy$(
+    strategy: RxStrategyNames<string> | Observable<RxStrategyNames<string>>
+  ) {
+    this.strategy.next(strategy);
+  }
+
+  @Output() readonly renderCallback = new Subject<void>();
```

Comment:
I'd like to chime in with `rxLet` directive, and notice that it has a [`rendered` output](https://github.com/rx-angular/rx-angular/blob/master/libs/template/let/src/lib/let.directive.ts#L363). I think these outputs should have a naming convetion with a prefix or something, or be consistent as Angular can handle [outputs with the same name](https://stackblitz.com/edit/angular-ivy-x25zxw?file=src%2Fapp%2Fapp.component.html).